internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -386,30 +398,32 @@ func (m *BlobMetadataManager) createFileFinalize(filePath string, fileMetadata [
 
 // GetFile reads and returns the content of metadata for a file.
 // TODO: Replace the two REST API calls with a single call to DownloadStream.
-func (m *BlobMetadataManager) getFile(filePath string) ([]byte, int64, error) {
+func (m *BlobMetadataManager) getFile(filePath string) ([]byte, int64, dcache.FileState, error) {
 	common.Assert(len(filePath) > 0)
 
 	path := filepath.Join(m.mdRoot, "Objects", filePath)
 	// Get the metadata content from storage.
 	data, prop, err := m.getBlobSafe(path)
 	if err != nil {
 		log.Debug("GetFile:: Failed to get metadata file content for file %s: %v", path, err)
-		return nil, -1, err
```

Comment:
pls add this

+               //
+               // getBlobSafe() should only fail when blob is non-existent.
+               // Assert to catch any other error.
+               //
+               common.Assert(errors.Is(err, syscall.ENOENT), err)